### PR TITLE
Fix/send responses everywhere customInputs were being sent.

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Adds responses everywhere customInputs was being sent earlier

**Environment**:  Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Test different scenarios like Request Reschedule(Email, Calendar, WEbhook) Reschedule(Email, Calendar, WEbhook), Cancellation(Email, Calendar, WEbhook), Reminder Emails

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/lib/getBooking.tsx
Comment (hariombalhara) [inline packages/lib/getBooking.tsx:17]: wasn't needed

File: packages/trpc/server/routers/viewer/bookings.tsx
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/bookings.tsx:783]: No need to parse here. getCalEventResponses can handle the parsing itself now.

File: packages/features/bookings/lib/getCalEventResponses.ts
Comment (hariombalhara) [inline packages/features/bookings/lib/getCalEventResponses.ts:80]: It allows us to simply spread over the return value to build cal event.

**Review by CarinaWolli — COMMENTED** [review]
It's also missing for the endpoint to test our zapier trigger. It's this endpoint: https://github.com/calcom/cal.com/blob/main/packages/app-store/zapier/api/subscriptions/listBookings.ts

Could you add that too?

**Comment by hariombalhara** [comment]
@CarinaWolli Thanks !! I have added it at that place also now.

File: pr-progress/pr-progress.md
Comment (CarinaWolli) [inline pr-progress/pr-progress.md:1]: we should remove that file before merging, @hariombalhara can you make sure that the found bugs are not getting lost and issues are created for them?

File: packages/features/webhooks/lib/sendPayload.ts
Comment (CarinaWolli) [inline packages/features/webhooks/lib/sendPayload.ts:49]: I pushed that change. We need to have the same payload in the zapier payload and in the /listBookings endpoint

**Review by CarinaWolli — COMMENTED** [review]
LGTM! I pushed a change to add userFieldsResponses to the Zapier payload. Will approve after pr-progress.md is removed

File: pr-progress/pr-progress.md
Comment (hariombalhara) [reply pr-progress/pr-progress.md]: Done and filed bugs. One of the fixes got merged today itself.

**Review by CarinaWolli — APPROVED** [review]
Nice, looking good now 🙌🏻

**Pull request merged**
